Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -891,6 +891,131 @@ final class DeclarationTests: ParserTestCase {
     )
   }
 
+  func testTypedThrows() {
+    assertParse(
+      "func test() throws(any Error) -> Int { }",
+      experimentalFeatures: [.typedThrows]
+    )
+
+    assertParse(
+      """
+      struct X {
+        init() throws(any Error) { }
+      }
+      """,
+      experimentalFeatures: [.typedThrows]
+    )
+
+    assertParse(
+      "func test() throws(MyError) 1️⃣async {}",
+      diagnostics: [
+        DiagnosticSpec(message: "'async' must precede 'throws'", fixIts: ["move 'async' in front of 'throws'"])
+      ],
+      fixedSource: "func test() async throws(MyError) {}",
+      experimentalFeatures: [.typedThrows]
+    )
+    assertParse(
+      "func test() throws 1️⃣async2️⃣(MyError) {}",
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "'async' must precede 'throws'", fixIts: ["move 'async' in front of 'throws'"]),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "unexpected code '(MyError)' in function"),
+      ],
+      fixedSource: "func test() async throws (MyError) {}",
+      experimentalFeatures: [.typedThrows]
+    )
+    assertParse(
+      "func test() 1️⃣try2️⃣(MyError) async {}",
+      diagnostics: [
+        DiagnosticSpec(locationMarker: "1️⃣", message: "expected throwing specifier; did you mean 'throws'?", fixIts: ["replace 'try' with 'throws'"]),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "unexpected code '(MyError) async' in function"),
```

Comment:
I'm surprised this isn't handled as it is above when `try` is the proper `throws` or below where `try` is before `async`, ie. that `async` is unexpected here.

Reply:
It's because I am not parsing the thrown type clause along a recovery path, I think.